pkg/auth/data/authconfig_data.go

```diff
@@ -103,8 +108,42 @@ func addAuthConfigCore(name, aType string, enabled, sloSupported bool, managemen
 		Enabled:            enabled,
 		LogoutAllSupported: sloSupported,
 	})
-	if err != nil && !apierrors.IsAlreadyExists(err) {
-		return err
+	if err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+
+		// The AC already exists. The result of the Create (in `createdOrKnown`) is just the
+		// object given to it as argument. It is NOT the object found in the store. It is
+		// still needed as argument for Patch().
+		//
+		// Given that a Patch operation is (a) quick, and (b) does not throw an error when
+		// given a no-op, pulling the existing object and checking if the flag has the
+		// proper value or not is not really an optimization. Setting the flag is therefore
+		// done unconditionally. This ensures both existence and proper value without fuss.
+		//
+		// Note that the `add` operation works even when the flag field exists. It behaves
+		// like a `replace`. This may just be because the flag is a scalar field, neither
+		// object, nor array.
+
```

Comment:
nit: I'd suggest not to use abbreviations, besides common/well-known ones, in comments. It took me a second to understand what you referred to as `AC` here.
Also it might be worth considering a more succinct version of the comment here. Something along the lines of 
```suggestion
		// Make sure the logoutAllSupported field is set correctly for the existing authConfig.
		// Use patch to avoid fetching the object first.
```